discovery/azure/azure.go

```diff
@@ -569,7 +569,13 @@ func (client *azureClient) getScaleSetVMs(ctx context.Context, scaleSet armcompu
 }
 
 func mapFromVM(vm armcompute.VirtualMachine) virtualMachine {
-	osType := string(*vm.Properties.StorageProfile.OSDisk.OSType)
+	var osType string
+	if vm.Properties != nil &&
```

Comment:
Move this inside the existing `if vm.Properties != nil` (previously on line 582)